### PR TITLE
Add tests for yandex hook [part of #8278]

### PR DESCRIPTION
**Pull request by subkanthi**
Paart of: https://github.com/apache/airflow/issues/8278
---
Make sure to mark the boxes below before creating PR: [x]

- [ x] Description above provides context of the change
- [ x] Unit tests coverage for changes (not needed for documentation changes)
- [ x] Target Github ISSUE in description if exists
- [ x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ x] Relevant documentation is updated including usage instructions.
- [ x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by potiuk** [comment]
Looks good! But I think you need to remove the yandex from list of those files without tests :)

**Comment by mik-laj** [comment]
@peter-volkov  Can you look at it?

**Comment by peter-volkov** [comment]
@potiuk what list are you talking about?
@mik-laj looks good
what can I do to improve the tests?

File: tests/providers/yandex/hooks/test_yandex.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import mock
+
+from airflow.exceptions import AirflowException
+from airflow.providers.yandex.hooks.yandex import YandexCloudBaseHook
+
+
+class TestYandexHook(unittest.TestCase):
+
+    @mock.patch('airflow.hooks.base_hook.BaseHook.get_connection')
+    @mock.patch('airflow.providers.yandex.hooks.yandex.YandexCloudBaseHook._get_credentials')
```
Comment (mik-laj) [inline tests/providers/yandex/hooks/test_yandex.py:29]: get_credentials method contains important logic for this class. What do you think to complete the tests for it? Now this method is simple, but in the future new authentication methods may be needed to be added. Tests will allow easy refactoring and prevent regression.

**Comment by mik-laj** [comment]
> @potiuk what list are you talking about?

https://github.com/apache/airflow/blob/master/tests/test_project_structure.py#L45

> what can I do to improve the tests?

If you want you can leave a comment. You are an YandexCloud expert, so I thought it worth giving you the chance to express your opinion about it

File: tests/providers/yandex/hooks/test_yandex.py (hunk 1 above)
Comment (peter-volkov) [reply tests/providers/yandex/hooks/test_yandex.py:29]: Ok. I will do it a bit later. I have plans to add some new features to this provider. I will include a test to the another PR.

File: tests/providers/yandex/hooks/test_yandex.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+import mock
```
Comment (mik-laj) [inline tests/providers/yandex/hooks/test_yandex.py]:
```suggestion
from unittest import mock
```

Comment (subkanthi) [reply tests/providers/yandex/hooks/test_yandex.py]: @mik-laj  Fixed this, is there anything else required to merge this PR, please let me know.

Comment (mik-laj) [reply tests/providers/yandex/hooks/test_yandex.py]: As mentioned before - https://github.com/apache/airflow/pull/9665#issuecomment-653931518, you still have to delete this file from the list of missing tests. This list is covered by unit test. We must have green CI to be able to merge this change.
```
tests/test_logging_config.py ........                                    [ 16%]
tests/test_project_structure.py .F

___________ TestProjectStructure.test_keep_missing_test_files_update ___________

self = <tests.test_project_structure.TestProjectStructure testMethod=test_keep_missing_test_files_update>

    def test_keep_missing_test_files_update(self):
        new_test_files = []
        for test_file in MISSING_TEST_FILES:
            if os.path.isfile(f"{ROOT_FOLDER}/{test_file}"):
                new_test_files.append(test_file)
        if new_test_files:
            new_files_text = '\n'.join(new_test_files)
            self.fail(
>               "You've added a test file currently listed as missing:\n"
                f"{new_files_text}"
                "\n"
                "Thank you very much.\n"
                "Can you remove it from the list of missing tests, please?"
            )
E           AssertionError: You've added a test file currently listed as missing:
E           tests/providers/yandex/hooks/test_yandex.py
E           Thank you very much.
E           Can you remove it from the list of missing tests, please?

tests/test_project_structure.py:117: AssertionError
```

Comment (subkanthi) [reply tests/providers/yandex/hooks/test_yandex.py]: thanks will do that

Comment (subkanthi) [reply tests/providers/yandex/hooks/test_yandex.py]: @mik-laj removed the file from the list and I see green CI. Thanks ready for merge.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**